sdk/core/AzureCore/Source/Util/ReachabilityManager.swift

```diff
@@ -99,7 +100,15 @@ import Foundation
         // MARK: - Listening
 
         public func registerListener(_ listener: @escaping ReachabilityStatusListener) {
-            self.listener = listener
+            self.listener = { status in
+                // no nothing on the initial status update to force the developer to resume transfers
+                // TODO: Maybe this should be set in a more thoughtful manner
+                guard self.initialized else {
+                    self.initialized = true
+                    return
```

Comment:
Initialized flag simply serves to skip the initial listener call that establishes the initial connectivity state.

Reply:
If this works, I'm okay with it for now. I think perhaps the "proper" way to do this would be to store the current state somewhere (initially "unknown") and if the previous state was unknown, don't call the callback but only store the new state.

Reply:
It was a quick fix and I liked your approach better. I also make sure that the status did actually change.